server/src/test/java/org/opensearch/search/DeletePitMultiNodeTests.java

```diff
@@ -0,0 +1,286 @@
+            }
+        });
+
+        ensureGreen();
+        /**
+         * When we invoke delete again, returns success after clearing the remaining readers. Asserting reader context
+         * not found exceptions don't result in failures ( as deletion in one node is successful )
+         */
+        ActionFuture<DeletePitResponse> execute = client().execute(DeletePitAction.INSTANCE, deletePITRequest);
+        DeletePitResponse deletePITResponse = execute.get();
+        for (DeletePitInfo deletePitInfo : deletePITResponse.getDeletePitResults()) {
+            assertTrue(pitIds.contains(deletePitInfo.getPitId()));
+            assertTrue(deletePitInfo.isSuccessful());
+        }
+        client().admin().indices().prepareDelete("index1").get();
+    }
+
+    public void testDeleteAllPitsWhileNodeDrop() throws Exception {
+        createPitOnIndex("index");
+        createIndex("index1", Settings.builder().put("index.number_of_shards", 5).put("index.number_of_replicas", 1).build());
+        client().prepareIndex("index1").setId("1").setSource("field", "value").setRefreshPolicy(IMMEDIATE).execute().get();
+        ensureGreen();
+        DeletePitRequest deletePITRequest = new DeletePitRequest("_all");
+        internalCluster().restartRandomDataNode(new InternalTestCluster.RestartCallback() {
+            @Override
+            public Settings onNodeStopped(String nodeName) throws Exception {
+                ActionFuture<DeletePitResponse> execute = client().execute(DeletePitAction.INSTANCE, deletePITRequest);
+                try {
+                    DeletePitResponse deletePITResponse = execute.get();
+                    for (DeletePitInfo deletePitInfo : deletePITResponse.getDeletePitResults()) {
+                        assertThat(deletePitInfo.getPitId(), not(blankOrNullString()));
+                        assertFalse(deletePitInfo.isSuccessful());
+                    }
+                } catch (Exception e) {
+                    assertTrue(e.getMessage().contains("Node not connected"));
+                }
+                return super.onNodeStopped(nodeName);
+            }
+        });
+
+        ensureGreen();
+        /**
+         * When we invoke delete again, returns success after clearing the remaining readers. Asserting reader context
+         * not found exceptions don't result in failures ( as deletion in one node is successful )
+         */
+        ActionFuture<DeletePitResponse> execute = client().execute(DeletePitAction.INSTANCE, deletePITRequest);
+        DeletePitResponse deletePITResponse = execute.get();
+        for (DeletePitInfo deletePitInfo : deletePITResponse.getDeletePitResults()) {
+            assertThat(deletePitInfo.getPitId(), not(blankOrNullString()));
+            assertTrue(deletePitInfo.isSuccessful());
+        }
+        client().admin().indices().prepareDelete("index1").get();
+    }
+
+    public void testDeleteWhileSearch() throws Exception {
```

Comment:
Lets add more concurrent tests to ensure all failure cases around grouped listener is covered